### PR TITLE
Add decorators closes #178

### LINKED ISSUE #178: Move Decorators written for tutorial to ART
in the first tutorial we introduce few decorators. Why don't add them to the ART directly and let user use them straightaway. In V0.4 we can write even more of them.

1.  LogginDecorator - this will just log some basic statistics of tensors inputted to the network
2. EvaluationSaver - this will show logits changes with time
3. Batch Saver - this will just save images before they are inputted to the network

It's just about moving them from tutorial to art not implementing them

### PR DESCRIPTION
**Pull request by SebChw**
Previously we had to implement decorators on our own + we decorated classes.

```python
def first_logging_func(data):
    logging.warning(data)

art_decorate([(MNISTModel, "predict")], first_logging_func)
```

Decorating class is an awful idea as function persists decorated during all runs. For scripts it's not that big deal but for notebooks much bigger.

Now decorating model work like this

```python
from art.decorators import ModelDecorator, LogInputStats
project.run_all(model_decorators=[ModelDecorator("predict", LogInputStats())])
```

So:

1. Some decorators are provided to the users
2. Decorators persist only within specific project run

### CODE REVIEW

**Comment by trebacz626** [comment]
Am I right that log inputs stats just prints inputs?

File: art/decorators.py [hunk 1]
```diff
@@ -66,7 +78,101 @@ def art_decorate(
...
+
+class BatchSaver:
+    """Save images from batch to debug_images folder"""
+
+    def __init__(self, how_many_batches=10, image_key_name=INPUT):
+        """
+        Args:
+            how_many_batches (int, optional): How many batches to save. Defaults to 10.
+            image_key_name (str, optional): under what . Defaults to "input".
+            image_key_idx (int, optional): _description_. Defaults to 0.
```
Comment (kordc) [inline art/decorators.py]: I thinks it's no longer here

**Review by kordc — COMMENTED** [review]
Besides this small doc mistake - LGTM

**Comment by kordc** [comment]
The link from the title hasn't applied, so just for reference - #178

**Comment by SebChw** [comment]
> Am I right that log inputs stats just prints inputs?

Yes, but at first it turn on lovely tensors. So, mean, min-max and std are printed

**Review by kordc — APPROVED** [review]
